### PR TITLE
add contract dev track to the documentation

### PR DESCRIPTION
**Pull request by iurimatias**
choose_your_adventure

### CODE REVIEW

**Comment by andytudhope** [comment]
This is a sh*t tonne of documentation 😅 Really awesome work @iurimatias! I don't have time today to do a full review, but perhaps you can give me some insight into your intentions with all this and how you want to push documentation/video efforts over the next few months?

File: scripts/helpers.js [hunk 1]
```diff
@@ -13,6 +13,11 @@ function startsWith(str, start) {
   return str.substring(0, start.length) === start;
 }
 
+hexo.extend.helper.register('is_active_track', function(track) {
+  var type = this.page.canonical_path.split('/')[0];
+	return (type === track ? 'active': '');
+});
```
Comment (alaibe) [inline scripts/helpers.js]: 2 spaces to indent?

File: source/docs_contracts/console_commands.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+* ``webserver stop`` - stop the dev webserver
+* ``browser open`` - open a web browser and load your DApp from the dev webserver
+
+### Custom Commands
+
+It's possible to extend Embark to include custom commands. See [how to create
+a plugin](creating_plugins.html)
+
+<pre><code class="javascript">module.exports = function(embark) {
+        embark.registerConsoleCommand(function(cmd, options) {
```
Comment (alaibe) [inline source/docs_contracts/console_commands.md]: I believe this is different now

File: source/docs_contracts/contributing.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+6. Create a pull request and describe the change.
+
+### Notice
+
+- Don't modify version number in `package.json`.
+- Your pull request will only get merged when tests passed. Don't forget to run tests before submission.
+
+    {% code %}
+    $ npm run fulltest
```
Comment (alaibe) [inline source/docs_contracts/contributing.md]: It should be `npm run test` I believe

File: source/docs_contracts/contributing.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+## Reporting Issues
+
+When you encounter some problems when using Embark, you can find the solutions in [Troubleshooting](troubleshooting.html) or ask me on [GitHub](https://github.com/embark-framework/embark/issues). If you can't find the answer, please report it on GitHub.
+
+1. Represent the problem in [debug mode](commands.html#Debug_mode).
+2. Run `embark version` and check the version info.    
+3. Post both debug message and version info on GitHub.
+
+[embark-framework/embark]: https://github.com/embark-framework/embark
```
Comment (alaibe) [inline source/docs_contracts/contributing.md]: Should this be a link to a new issue?

File: source/docs_contracts/embark_commands.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+The `--node` option supports several values:
+
+Value | Description
+--- | ---
+`vm` | start and use an Ethereum simulator (ganache)
+`embark` | use the node of a running embark process
+`<endpoint>` | connect to and use the specified node
+
+Example of endpoint usage: `embark test --node ws://localhost:8556`
```
Comment (alaibe) [inline source/docs_contracts/embark_commands.md]: Why not using the port 8546.
The fact that there is proxy is not obvious and in the conf by default it is 8546

File: source/docs_contracts/index.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+title: Getting Started
+---
+Welcome to the Embark documentation! The following guides will help you guiding your way through your journey of building decentralized applications. If you encounter any problems when using Embark, or find anything unclear described here, please have a look at the [troubleshooting guide](troubleshooting.html), raise an issue on [GitHub](https://github.com/embark-framework/embark/issues) or ask in the [Gitter Chatroom](https://gitter.im/embark-framework/Lobby) respectively.
```
Comment (alaibe) [inline source/docs_contracts/index.md]: Do we already have a stackoverflow?

**Pull request closed without merging**